### PR TITLE
Implement Contest Logo Override

### PR DESCRIPTION
**Pull request by WallE256**
Addresses #718

### CODE REVIEW

File: judge/jinja2/camo.py [hunk 1]
```diff
@@ -0,0 +1,7 @@
+from judge.utils.camo import client as camo_client
+
+from . import registry
+
+@registry.function
```
Comment (quantum5) [inline judge/jinja2/camo.py]: I think a filter is better than a function.

File: templates/site-logo-fragment.html [hunk 2]
```diff
@@ -1,2 +1,6 @@
+{% if request.in_contest and request.participation.contest.logo_override %}
+<img src="{{ camo(request.participation.contest.logo_override) }}" alt="{{ SITE_NAME }}" height="44" style="border: none">
```
Comment (quantum5) [inline templates/site-logo-fragment.html]: Indent and switch to filter.

File: templates/site-logo-fragment.html [hunk 3]
```diff
@@ -1,2 +1,6 @@
...
+{% else %}
 <img src="{{ static('icons/logo.svg') }}" alt="{{ SITE_NAME }}" width="160" height="44"
```
Comment (quantum5) [inline templates/site-logo-fragment.html]: Indent

File: judge/models/contest.py [hunk 4]
```diff
@@ -84,6 +84,7 @@ class Contest(models.Model):
     organizations = models.ManyToManyField(Organization, blank=True, verbose_name=_('organizations'),
                                            help_text=_('If private, only these organizations may see the contest'))
     og_image = models.CharField(verbose_name=_('OpenGraph image'), default='', max_length=150, blank=True)
+    logo_override = models.CharField(verbose_name=_('Logo Override'), default='', max_length=150, blank=True)
```
Comment (Xyene) [inline judge/models/contest.py]: "Override" should be lowercase. Also missing help text. A better name would be `logo_override_image`.

File: templates/site-logo-fragment.html [hunk 5]
```diff
@@ -1,2 +1,6 @@
+{% if request.in_contest and request.participation.contest.logo_override %}
```
Comment (Xyene) [inline templates/site-logo-fragment.html]: Should also be shown on the `/contest/` page itself, even if you're outside of the contest.

File: judge/jinja2/__init__.py [hunk 6]
```diff
@@ -9,7 +9,7 @@
 from judge.highlight_code import highlight_code
 from judge.user_translations import ugettext
 from . import (datetime, filesize, gravatar, language, markdown, rating, reference, render, social, spaceless,
-               submission, timedelta)
+               submission, timedelta, camo)
```
Comment (Xyene) [inline judge/jinja2/__init__.py]: Fix import order.

File: judge/jinja2/camo.py [hunk 7]
```diff
@@ -0,0 +1,7 @@
+from judge.utils.camo import client as camo_client
+
+from . import registry
+
```
Comment (Xyene) [inline judge/jinja2/camo.py]: I think there's supposed to be two lines here.

File: templates/site-logo-fragment.html [hunk 8]
```diff
@@ -1,2 +1,8 @@
-<img src="{{ static('icons/logo.svg') }}" alt="{{ SITE_NAME }}" width="160" height="44"
-     onerror="this.src=&quot;{{ static('icons/logo.png') }}&quot;; this.onerror=null" style="border: none">
+{% if request.in_contest and request.participation.contest.logo_override_image %}
+    <img src="{{ request.participation.contest.logo_override_image|camo }}" alt="{{ SITE_NAME }}" height="44" style="border: none">
+{% elif contest is defined and contest.logo_override_image %}
+    <img src="{{ contest.logo_override_image|camo }}" alt="{{ SITE_NAME }}" height="44" style="border: none">
+{% else %}
+    <img src="{{ static('icons/logo.svg') }}" alt="{{ SITE_NAME }}" width="160" height="44"
```
Comment (Xyene) [inline templates/site-logo-fragment.html]: Misaligned

File: judge/models/contest.py [hunk 9]
```diff
@@ -84,6 +84,8 @@ class Contest(models.Model):
     organizations = models.ManyToManyField(Organization, blank=True, verbose_name=_('organizations'),
                                            help_text=_('If private, only these organizations may see the contest'))
     og_image = models.CharField(verbose_name=_('OpenGraph image'), default='', max_length=150, blank=True)
+    logo_override_image = models.CharField(verbose_name=_('Logo override image'), default='', max_length=150, blank=True,
+                                           help_text=_('This image will replace the default site logo for users inside the contest.'))
```
Comment (Xyene) [inline judge/models/contest.py]: No ending . to be consistent with other messages here

Comment (WallE256) [reply judge/models/contest.py]: All the other help texts in this model have . after them.

**Pull request merged**